### DISCUSSION TITLE
slash boot slash config.txt

### DISCUSSION THREAD

**Discussion by alabamatoy** [jgyates/genmon#1083] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

Genmon: 1.19.02
OS Ubuntu 22.04 on RPi3

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

when I try to run serialconfig.py I get:
Error: unable to find file /boot/config.txt

**Comment by jgyates** [comment]
Is this a 64 bit versions of Ubuntu? It looks like the latest version of Raspbian 64 changed the location of the config.txt file from /boot to /boot/firmware. I assume this is the reason it is different on Ubuntu as well. I just checked a pi that has 64 bit Raspbian bookworm and the file /boot/config.txt has this message in it:

     DO NOT EDIT THIS FILE
     The file you are looking for has moved to /boot/firmware/config.txt

To date I have only tested with Raspbian, but this does appear to be a recent change with all debian based distros for the pi.

I should have a fix in the next couple of days, but if you want to work around this issue you then do this:

Back up your /boot/firmware/config.txt and the /boot/firmware/cmdline.txt files

      sudo cp /boot/firmware/config.txt /boot/firmware/config.txt.bakup
       sudo cp /boot/firmware/cmdline.txt /boot/firmware/cmdline.txt.bakup

Make a copy of /boot/firmware/config.txt and cmdline.txt and put it in the /boot folder

    sudo cp /boot/firmware/config.txt /boot/config.txt
    sudo cp /boot/firmware/cmdline.txt /boot/config.txt

Then run the script

Then copy the file back to /boot/firmware/config.txt, that will probably work.

     sudo cp /boot/config.txt /boot/firmware/config.txt
     sudo cp /boot/cmdline.txt /boot/firmware/cmdline.txt

Let me know if you have any questions or if you just prefer to wait for a fix.

**Reply by alabamatoy** [reply]
"Is this a 64 bit versions of Ubuntu?"

It is.  I'm in no rush, I'll wait for a fix.  Thanks for what you do.

**Comment by jgyates** [comment, marked as answer]
I checked in an update. You can update to the latest with this command:

     ./genmonmaint.sh -u

This should work on both old an new version of the firmware. serialconfig.py will print the location of the file it is working on. I tested with Raspberry pi OS, not Ubuntu, but the file locations appear to be the same. These files are read by the firmware, not the OS so it should work the same on both Raspbian and Ubuntu. Let me know how your testing goes.

**Reply by alabamatoy** [reply]
I ran genmonmaint.sh -u, got:
```
ubuntu@ubuntu:~$ sudo /genmon/./genmonmaint.sh -u
This script will update genmon to the latest version from the github repository. This script requires internet access. If you have modified any files in the genmon directory, they will be overwritten. Configuration files in the configuration directory will not be overritten.  Continue? (y/n)  y
Updating genmon...
/etc/genmon/update.txt does not exist. Creating..
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 10 (delta 7), reused 8 (delta 5), pack-reused 0
Unpacking objects: 100% (10/10), 1.43 KiB | 23.00 KiB/s, done.
From https://github.com/jgyates/genmon
   4eb1b5e..8d1687f  master     -> origin/master
HEAD is now at 8d1687f V1.19.02
Done.
```

Hmmm...version didnt change.  Did I fubar? Then I restarted genmon, no issues (other than the serial issue Im trying to resolve).  I stopped genmon, then ran serialconfig:
```
ubuntu@ubuntu:/$ sudo python3 /genmon/OtherApps/serialconfig.py -e

Using files: /boot/firmware/config.txt and /boot/firmware/cmdline.txt
Error: unable to find file /boot/config.txt
ubuntu@ubuntu:/$
```
??

**Reply by jgyates** [reply]
Sorry, I omitted on line that needed to be there. I updated the file so update again and you should be OK.

The version number will not change until I am ready to roll out any incremental changes to everyone running genmon. Once I update the version every will get an email.

Let me know how it goes.

**Reply by alabamatoy** [reply]
```
ubuntu@ubuntu:/$ sudo python3 /genmon/OtherApps/serialconfig.py -e

Using files: /boot/firmware/config.txt and /boot/firmware/cmdline.txt

Model: Raspberry Pi 3 Model B Rev 1.2

Enable UART: Success
Change BT serial port: Success
Disable BT: Success
Disable serial console: Success
Disable serial console service: Success
Removed /etc/systemd/system/multi-user.target.wants/hciuart.service.
Disable BT service service: Success

Serial port settings are OK. A reboot is needed if changes were made.

ubuntu@ubuntu:/$ 
```
Rebooted, no genmon.  Running it manually, its crashing because incorrect serial device...

```
ubuntu@ubuntu:~$ sudo /bin/bash /genmon/startgenmon.sh start
Starting genmon python scripts
Starting....
Starting /genmon/genmon.py 
Stopping genserv.py
Starting /genmon/genserv.py 
ubuntu@ubuntu:~$ 
```
genmon log shows:

```
2024-03-15 11:22:57,911 : Selected Controller: generac_evo_nexus
2024-03-15 11:22:57,933 : Error opening modbus device: Error opening modbus device. : generac_evolution.py:325
2024-03-15 11:22:57,933 : Possible cause is invalid serial port specified.
2024-03-15 11:34:32,634 : Selected Controller: generac_evo_nexus
2024-03-15 11:34:32,646 : Error opening modbus device: Error opening modbus device. : generac_evolution.py:325
2024-03-15 11:34:32,646 : Possible cause is invalid serial port specified.

```
It ran before, but had no generator data.  Now it wont even start.  I don't know how to change the serial port without the web GUI.

serialtest.py looks at /dev/serial0, but I had /dev/ttyS0 in settings.  Im starting to think I have fubarred directory structure somehow, I find 2 genmon.conf files, one in /genmon/conf and one in /etc/genmon

Sorry to be such a PITA.

**Reply by jgyates** [reply]
All of the software modules are documented here: https://github.com/jgyates/genmon/wiki/1----Software-Overview#otherappsserialtestpy-optional

serialtest.py defaults to /dev/serial0 but the will take any port on the command line.

the files in /etc/genmon are the run time versions. All the files in ~/genmon (including ~/genmon/conf) are updated via the git command so a software update will overwrite the ~/genmon/conf files. The install script copies the files to /etc/genmon

Sounds like you just need to test your serial port and make sure you genmon settings for serial are valid. Ubuntu may name the serial port differently, i don't really know as I don't teat with Ubuntu. The install script will put /dev/serial0 in the /etc/genmon/genon.conf file so if Ubuntu is not using /dev/serial0 for the onboard serial port you will need ot change that either by editing the /etc/genmon/genon.conf file or via the setttings page in the web interface.

i would test the serial port per these instructions:

https://github.com/jgyates/genmon/wiki/3.2--Setup-Your-Pi-and-Serial-Port

Out of curiosity, is there are reason you are using Ubuntu instead of Raspberry pi OS?

**Reply by alabamatoy** [reply]
"Out of curiosity, is there are reason you are using Ubuntu instead of Raspberry pi OS?"  Familiarity....

OK, I modified /etc/genmon/genmon.conf to point at /dev/ttyAMA0, reran startgenmon.sh and it immediately began working.

Your help is greatly appreciated.  Now to figure out how to make it start on boot....

**Comment by jgyates** [comment]
the install script gives you and option to add genmon to the crontab. You can manually do this by this command:

    
     sudo crontab -e

this will bring up a text editor to edit the crontab, then add this line:

     @reboot sleep 30 && /bin/bash /home/pi/genmon/startgenmon.sh start

Note that you may need to change the path to the startgenmon.sh script, depending on where you installed genmon. Save the file. Then it will start on every reboot after a 30 second delay (the delay is to allow the network to come up)
